pkg/cloudprovider/providers/gce/gce_loadbalancer_internal.go

```diff
@@ -126,39 +141,25 @@ func (gce *GCECloud) ensureInternalLoadBalancer(clusterName, clusterID string, s
 	return status, nil
 }
 
-func (gce *GCECloud) clearExistingInternalLB(loadBalancerName string, existingFwdRule, expectedFwdRule *compute.ForwardingRule, expectedBSName string) (fwdRuleDeleted bool, err error) {
-	if existingFwdRule == nil {
-		return false, nil
-	}
-
-	if !fwdRuleEqual(existingFwdRule, expectedFwdRule) {
-		glog.V(2).Infof("clearExistingInternalLB(%v: deleting existing forwarding rule with IP address %v", loadBalancerName, existingFwdRule.IPAddress)
-		if err = gce.DeleteRegionForwardingRule(loadBalancerName, gce.region); err != nil && !isNotFound(err) {
-			return false, err
+func (gce *GCECloud) clearPreviousInternalResources(loadBalancerName string, existingBackendService *compute.BackendService, expectedBSName, expectedHCName string) error {
+	if existingBackendService.Name != expectedBSName {
+		glog.V(2).Infof("clearPreviousInternalResources(%v): expected backend service %q does not match previous %q - deleting backend service", loadBalancerName, expectedBSName, existingBackendService.Name)
+		if err := gce.teardownInternalBackendService(existingBackendService.Name); err != nil && !isNotFound(err) {
+			glog.Warningf("clearPreviousInternalResources: could not delete old backend service: %v, err: %v", existingBackendService.Name, err)
 		}
-		fwdRuleDeleted = true
 	}
 
-	existingBSName := getNameFromLink(existingFwdRule.BackendService)
-	bs, err := gce.GetRegionBackendService(existingBSName, gce.region)
-	if err == nil {
-		if bs.Name != expectedBSName {
-			glog.V(2).Infof("clearExistingInternalLB(%v): expected backend service %q does not match actual %q - deleting backend service & healthcheck & firewall", loadBalancerName, expectedBSName, bs.Name)
-			// Delete the backend service as well in case it's switching between shared, nonshared, tcp, udp.
-			var existingHCName string
-			if len(bs.HealthChecks) == 1 {
-				existingHCName = getNameFromLink(bs.HealthChecks[0])
-			}
-			if err = gce.teardownInternalBackendResources(existingBSName, existingHCName); err != nil {
-				glog.Warningf("clearExistingInternalLB: could not delete old resources: %v", err)
-			} else {
-				glog.V(2).Infof("clearExistingInternalLB: done deleting old resources")
+	if len(existingBackendService.HealthChecks) == 1 {
```

Comment:
Looked at the GCP docs. 

> The list of URLs to the HttpHealthCheck or HttpsHealthCheck resource for health checking this BackendService. Currently at most one health check can be specified, and a health check is required. 

I'll log a warning, but as of now, it's impossible.